.github/workflows/mega-linter-for-runner.yml

```diff
@@ -113,61 +258,91 @@ jobs:
+      # This doesn't currently work for forks, only Pull Requests from the same
+      # repository will have a new Pull Request created.
+      #
       - name: Create Pull Request with applied fixes
         uses: peter-evans/create-pull-request@v5
         id: cpr
-        if: env.APPLY_FIXES_IF_PR == 'true'
+        if: >-
+          steps.ml.outputs.has_updated_sources == 1 &&
+          (
+            env.APPLY_FIXES_EVENT == 'all' ||
+            env.APPLY_FIXES_EVENT == github.event_name
+          ) &&
+          env.APPLY_FIXES_MODE == 'pull_request' &&
+          (
+            github.event_name == 'push' ||
+            github.event.pull_request.head.repo.full_name == github.repository
+          ) &&
+          !contains(github.event.head_commit.message, 'skip fix')
         with:
           token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
           commit-message: "[MegaLinter] Apply linters automatic fixes"
           title: "[MegaLinter] Apply linters automatic fixes"
           labels: bot
 
-      - name: Create PR output
-        if: env.APPLY_FIXES_IF_PR == 'true'
+      - name: Create Pull Request output
+        if: >-
+          steps.ml.outputs.has_updated_sources == 1 &&
+          (
+            env.APPLY_FIXES_EVENT == 'all' ||
+            env.APPLY_FIXES_EVENT == github.event_name
+          ) &&
+          env.APPLY_FIXES_MODE == 'pull_request' &&
+          (
+            github.event_name == 'push' ||
+            github.event.pull_request.head.repo.full_name == github.repository
+          ) &&
+          !contains(github.event.head_commit.message, 'skip fix')
         run: |
           echo "PR Number - ${{ steps.cpr.outputs.pull-request-number }}"
           echo "PR URL - ${{ steps.cpr.outputs.pull-request-url }}"
 
-      # Push new commit if applicable
-      # (for now works only on PR from same repository, not from forks)
+      ##
+      # Create a commit with any automatic fixes from linters that support
+      # fixing, if configured to do so via `APPLY_FIXES` and if
+      # `APPLY_FIXES_MODE` is set to `commit`.
+      #
+      # This doesn't currently work for forks, only Pull Requests from the same
+      # repository will have a new Pull Request created.
+      #
       - name: Prepare commit
-        if: env.APPLY_FIXES_IF_COMMIT == 'true'
+        if: >-
+          steps.ml.outputs.has_updated_sources == 1 &&
+          (
+            env.APPLY_FIXES_EVENT == 'all' ||
+            env.APPLY_FIXES_EVENT == github.event_name
+          ) &&
+          env.APPLY_FIXES_MODE == 'commit' &&
+          github.ref != 'refs/heads/main' &&
+          (
+            github.event_name == 'push' ||
+            github.event.pull_request.head.repo.full_name == github.repository
+          ) &&
+          !contains(github.event.head_commit.message, 'skip fix')
         run: sudo chown -Rc $UID .git/
 
       - name: Commit and push applied linter fixes
-        uses: stefanzweifel/git-auto-commit-action@v5
```

Comment:
Please freshen from main, as the action's version is being downgraded from `v5` to `v4`.